tools/backup_analysis.py

```diff
@@ -0,0 +1,174 @@
+        if buf1 != buf2:
+            total_bytes_diff += cmp_size
+            # Count how many granularity level blocks changed.
+            if bytes_gran == 0:
+                gran_blocks += 1
+            bytes_gran += cmp_size
+    # Account for anything from the final iteration of the loop and partial blocks.
+    if partial != 0:
+        buf1 = fp1.read(partial)
+        buf2 = fp2.read(partial)
+        # If they're different, gather information.
+        if buf1 != buf2:
+            total_bytes_diff += partial
+            bytes_gran += partial
+    fp1.close()
+    fp2.close()
+    end = time.asctime()
+    if bytes_gran != 0:
+        if offset != 0:
+            print("{}: Offset {}: {} bytes differ in {} bytes".format(filename, start_off, bytes_gran, granularity))
+        if bytes_gran <= pct20:
+            pct20_count += 1
+        elif bytes_gran >= pct80:
+            pct80_count += 1
+
+    # Report for each file.
+    print("{}: started {} completed {}".format(filename, start, end))
+    print("{} ({}) differs by {} bytes in {} granularity blocks".format(filename, file_size, total_bytes_diff, gran_blocks ))
+    if gran_blocks != 0:
+        pct20_blocks = round(abs(pct20_count / gran_blocks * 100))
+        pct80_blocks = round(abs(pct80_count / gran_blocks * 100))
+        print("{}: {} of {} blocks ({}%) differ by {} bytes or less of {} (20%)".format(filename, pct20_count, gran_blocks, pct20_blocks, pct20, granularity))
+        print("{}: {} of {} blocks ({}%) differ by {} bytes or more of {} (80%)".format(filename, pct80_count, gran_blocks, pct80_blocks, pct80, granularity))
+
+def compare_backups(dir1, dir2, granularity):
+    common=set()
+    files1=set(fnmatch.filter(os.listdir(dir1), "*.wt"))
+    files2=set(fnmatch.filter(os.listdir(dir2), "*.wt"))
+
+    common = files1.intersection(files2)
+    # For now assume the first directory is the older one. Once we add functionality to parse the
+    # text in WiredTiger.backup we can look at the checkpoint timestamp of a known table like the
+    # history store and figure out which is older.
+    for file in files1.difference(files2):
+        print(file + " dropped between backups")
+    for file in files2.difference(files1):
+        print(file + " created between backups")
+    print(common)
+    for f in common:
+        f1_size = os.stat(os.path.join(dir1, f)).st_size
+        f2_size = os.stat(os.path.join(dir2, f)).st_size
+        min_size = min(f1_size, f2_size)
+        # For now we're only concerned with changed blocks between backups.
+        # So only compare the minimum size both files have in common.
+        # FIXME: More could be done here to report extra blocks added/removed.
+        compare_file(dir1, dir2, f, min_size, 4096, granularity)
+
+def backup_analysis(args):
+    if len(args) < 2:
+        usage_exit()
+    dir1 = args[0]
+    dir2 = args[1]
+    if len(args) > 2:
+        granularity = int(args[2])
```

Comment:
The usage info implies that I should be able to write `16MB` as the third argument, but this doesn't seem to be the case. You may want to update the usage comment to indicate that this expects the number of bytes.

Reply:
Fixed. Good catch.